plugin-server/src/cdp/consumers/cdp-behavioural-events.consumer.ts

```diff
@@ -0,0 +1,203 @@
+    protected name = 'CdpBehaviouralEventsConsumer'
+    protected kafkaConsumer: KafkaConsumer
+
+    constructor(hub: Hub, topic: string = KAFKA_EVENTS_JSON, groupId: string = 'cdp-behavioural-events-consumer') {
+        super(hub)
+        this.kafkaConsumer = new KafkaConsumer({ groupId, topic })
+    }
+
+    public async processBatch(invocationGlobals: HogFunctionInvocationGlobals[]): Promise<void> {
+        return await this.runInstrumented('processBatch', async () => {
+            if (!invocationGlobals.length) {
+                return
+            }
+
+            // Track events consumed and matched (absolute numbers)
+            let eventsMatched = 0
+
+            for (const event of invocationGlobals) {
+                const matched = await this.processEvent(event)
+                if (matched) {
+                    eventsMatched++
+                }
+            }
+
+            // Update metrics with absolute numbers
+            counterEventsConsumed.inc(invocationGlobals.length)
+            counterEventsMatchedTotal.inc(eventsMatched)
+        })
+    }
+
+    private async processEvent(event: HogFunctionInvocationGlobals): Promise<boolean> {
+        try {
+            const teamId = event.project.id
+            const actions = await this.loadActionsForTeam(teamId)
+
+            if (!actions.length) {
+                logger.debug('No actions found for team', { teamId })
+                return false
+            }
+
+            let eventMatched = false
+            for (const action of actions) {
+                const matched = await this.doesEventMatchAction(event, action)
+                if (matched) {
+                    eventMatched = true
+                }
+            }
+
+            return eventMatched
+        } catch (error) {
+            logger.error('Error processing event', {
+                teamId: event.project.id,
+                eventName: event.event.event,
+                error,
+            })
+            return false
+        }
+    }
+
+    private async loadActionsForTeam(teamId: number): Promise<Array<{ id: string; name: string; bytecode: any }>> {
+        try {
+            const actions = await this.hub.actionManagerCDP.getActionsForTeam(teamId)
+            return actions.map((action) => ({
+                id: String(action.id),
+                name: action.name,
+                bytecode: action.bytecode,
+            }))
+        } catch (error) {
+            logger.error('Error loading actions for team', { teamId, error })
+            return []
+        }
+    }
+
+    private async doesEventMatchAction(
+        event: HogFunctionInvocationGlobals,
+        action: { id: string; name: string; bytecode: any }
+    ): Promise<boolean> {
+        if (!action.bytecode) {
+            return false
+        }
+
+        try {
+            // Convert event to filter globals format
+            const filterGlobals = convertToHogFunctionFilterGlobal(event)
```

Comment:
The back and forth from a CH event, to globals, to filter globals is super wasteful. Filters use almost exactly the same structure as the raw CH event so we should just have a nicer method to go straight to it